### PR TITLE
feat: add product listing

### PR DESCRIPTION
**Pull request by sahilshakya**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -3,23 +3,29 @@ import { BrowserRouter, Routes, Route } from "react-router-dom";
...
         <Routes>
-          <Route path="/" element={<Layout />}>
-            <Route path="/" element={<ProductPage />} />
-
+          <Route path="/" element={<MainLayout />}>
             <Route path="/login" element={<Login />} />
-            <Route path="/cart" element={<CartPage />} />
-            <Route path="*" element={<PageNotFound />} />
+
+            <Route path="/" element={<DashboardLayout />}>
```
Comment (AyushOmega) [inline src/App.tsx:23]: it's better to place routes in a constant file.

const uiRoutes = {
    cart: '/cart',
    home: '/',
    ....
}

File: src/App.tsx [hunk 2]
```diff
@@ -3,23 +3,29 @@ import { BrowserRouter, Routes, Route } from "react-router-dom";
...
+              <Route path="/" element={<ProductsPage />} />
+              <Route path="/singleproduct/:id" element={<Product />} />
```
Comment (AyushOmega) [inline src/App.tsx:25]: Better name to the route would be `/products/:id`

File: src/components/NavBar.tsx [hunk 3]
```diff
@@ -3,17 +3,35 @@ import H5 from "../shared/components/H5";
 import { BsFillCartFill, BsPersonFill } from "react-icons/bs";
 
 const NavBar = () => {
+  const token = localStorage.getItem("authToken");
```
Comment (AyushOmega) [inline src/components/NavBar.tsx:6]: make a common util (`storage.ts`) where you can place functions to getItem, setItem, clear items from localstorage.

File: src/components/NavBar.tsx [hunk 4]
```diff
@@ -3,17 +3,35 @@ import H5 from "../shared/components/H5";
...
       <Link to="/">
         <H5 styles="text-primary">NoveltyMart</H5>
       </Link>
 
-      <ul className="flex justify-normal items-center text-small font-bold text-gray-medium">
-        <Link to="/login" className="flex items-center mx-4">
-          <BsPersonFill className="mr-2" />
-          <p>Sign Up/Sign In</p>
-        </Link>
+      <ul className="flex justify-normal items-center text-normal font-bold text-gray-dark">
```
Comment (AyushOmega) [inline src/components/NavBar.tsx:13]: why `ul`? why not `div`?

File: src/components/NavBar.tsx [hunk 5]
```diff
@@ -3,17 +3,35 @@ import H5 from "../shared/components/H5";
...
+            </button>
+
+            <div className="absolute right-0 mt-2 w-48 bg-white rounded-md shadow-md ">
+              <a href="/login" className=" px-4 py-2 text-gray-dark ">
+                Logout
+              </a>
+            </div>
+          </div>
+        ) : (
+          <Link to="/login" className="flex items-center mx-4">
```
Comment (AyushOmega) [inline src/components/NavBar.tsx:29]: let's use value from constants.

`uiRoutes.login`

File: src/components/Protected.tsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { Navigate } from "react-router-dom";
+
+const Protected = ({ children }) => {
+  return <Navigate to="/" />;
```
Comment (AyushOmega) [inline src/components/Protected.tsx:4]: let's use value from constants.

uiRoutes.login

File: src/constant/api.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 export const BASE_URL = "http://localhost:5000/api/v1";
```
Comment (AyushOmega) [inline src/constant/api.ts:1]: API base url should reside in .env. Keep .env in .gitignore (DO NOT commit .env to remote branch).

File: src/login/pages/Login.tsx [hunk 8]
```diff
@@ -20,10 +22,18 @@ const Login: React.FC = () => {
   });
 
   const onSubmit = async (data: ILoginRequest) => {
-    const resp = await loginHandler({ data });
-    if (resp) navigate("/");
-    else {
-      console.log("Invalid credentials");
+    try {
+      const resp = await loginHandler({ data });
+      if (resp) {
```
Comment (AyushOmega) [inline src/login/pages/Login.tsx:27]: do we think this check for `resp`?

I think, it's adequate to handle error in the error catch.

File: src/products/pages/Product.tsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import { Link } from "react-router-dom";
+
+const Product = ({ product }) => {
+  return (
+    <Link to={`/singleproduct/${product.id}`}>
```
Comment (AyushOmega) [inline src/products/pages/Product.tsx:5]: take route from constant.

File: src/products/pages/Product.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+          <div className="px-4 py-3 w-72">
+            <p className="text-small font-regularBold ">
+              {product.brand[0].name}/{product.category.name}
+            </p>
+            <p className="text-normal font-regularBold  block capitalize">
+              {product.name}
+            </p>
+            <div className="flex items-center">
+              <p className="text-small font-bold  cursor-auto my-3">
+                ${product.price}
```
Comment (AyushOmega) [inline src/products/pages/Product.tsx:26]: formatting like these should be done in a separate layer after fetching data from API.

File: src/products/pages/ProductsPage.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  discontedPrice: number;
+  description: string;
+  images: [];
+}
+
+const ProductsPage = () => {
+  const [products, setProducts] = useState<Product[]>([]);
+
+  const getProducts = async () => {
+    const response = await axios.get("http://localhost:5000/api/v1/products");
```
Comment (AyushOmega) [inline src/products/pages/ProductsPage.tsx:19]: use a separate api layer.

**Review by AyushOmega — CHANGES_REQUESTED** [review]

**Comment by sahilshakya** [comment]
created new branch , with other changes

**Pull request closed without merging**